### PR TITLE
Add TGT macro for targeting data in json

### PR DESCRIPTION
**Pull request by bradfrizzell**
See https://github.com/ampproject/amphtml/issues/12374

### CODE REVIEW

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 1]
```diff
@@ -741,6 +741,10 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     return {
       PAGEVIEWID: () => Services.documentInfoForDoc(this.element).pageViewId,
       HREF: () => this.win.location.href,
+      TGT: () => {
+        const json = tryParseJson(this.element.getAttribute('data-json'));
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: TGT: () => (tryParseJson(this.element.getAttribute('data-json')) || {})['targeting'] || '',

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js [hunk 2]
```diff
@@ -294,6 +297,7 @@ describes.realWin('DoubleClick Fast Fetch RTC', {amp: true}, env => {
         'data-multi-size-validation': macros['data-multi-size-validation'],
         'data-override-width': macros['data-OVERRIDE-width'],
         'data-override-height': macros['data-override-HEIGHT'],
+        'data-json': JSON.stringify(json),
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js]: does your test coverage handle the case where data-json exists but without targeting key?

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js (hunk 1 above)
Comment (bradfrizzell) [reply extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: image

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js (hunk 2 above)
Comment (bradfrizzell) [reply extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js]: added test

**Comment by monish-chhadwa** [comment]
Hey, @bradfrizzell 
Should you be using this.element.dataset['json'] or this.element.getAttribute('json') ?

**Comment by bradfrizzell** [comment]
@monish-chhadwa Good catch, follow up PR to fix here: https://github.com/ampproject/amphtml/pull/12841

**Pull request merged**